app/add_note.py

```diff
@@ -0,0 +1,291 @@
+        person.source_date_local = self.to_local_time(person.source_date)
+        person.expiry_date_local = self.to_local_time(
+            person.get_effective_expiry_date())
+
+        # Get the notes and duplicate links.
+        try:
+            notes = person.get_notes()
+        except datastore_errors.NeedIndexError:
+            notes = []
+        person.sex_text = get_person_sex_text(person)
+        for note in notes:
+            self.add_fields_to_notes(note)
+        try:
+            linked_persons = person.get_all_linked_persons()
+        except datastore_errors.NeedIndexError:
+            linked_persons = []
+        linked_person_info = []
+        for linked_person in linked_persons:
+            try:
+                linked_notes = linked_person.get_notes()
+            except datastore_errors.NeedIndexError:
+                linked_notes = []
+            for note in linked_notes:
+                self.add_fields_to_notes(note)
+            linked_person_info.append(dict(
+                id=linked_person.record_id,
+                name=linked_person.primary_full_name,
+                view_url=self.get_url('/add_note', id=linked_person.record_id),
+                notes=linked_notes))
+
+        # Render the page.
+        dupe_notes_url = self.get_url(
+            '/add_note', id=self.params.id, dupe_notes='yes')
+        results_url = self.get_url(
+            '/results',
+            role=self.params.role,
+            query=self.params.query,
+            given_name=self.params.given_name,
+            family_name=self.params.family_name)
+        feed_url = self.get_url(
+            '/feeds/note',
+            person_record_id=self.params.id,
+            repo=self.repo)
+        subscribe_url = self.get_url('/subscribe', id=self.params.id)
+        delete_url = self.get_url('/delete', id=self.params.id)
+        disable_notes_url = self.get_url('/disable_notes', id=self.params.id)
+        enable_notes_url = self.get_url('/enable_notes', id=self.params.id)
+        extend_url = None
+        extension_days = 0
+        expiration_days = None
+        expiry_date = person.get_effective_expiry_date()
+        if expiry_date and not person.is_clone():
+            expiration_delta = expiry_date - get_utcnow()
+            extend_url =  self.get_url('/extend', id=self.params.id)
+            extension_days = extend.get_extension_days(self)
+            if expiration_delta.days < EXPIRY_WARNING_THRESHOLD:
+                # round 0 up to 1, to make the msg read better.
+                expiration_days = expiration_delta.days + 1
+
+        if person.is_clone():
+            person.provider_name = person.get_original_domain()
+
+        sanitize_urls(person)
+        for note in notes:
+            sanitize_urls(note)
+
+        if person.profile_urls:
+            person.profile_pages = get_profile_pages(person.profile_urls, self)
+
+        self.render('add_note.html',
```

Comment:
You can remove a lot of code from this file. For example, only record_id, notes_disabled and full_name properties are used for person object. We don't need to set any other fields e.g. profile_pages and provider_name. Other variables passed to add_note.html are also not used e.g. notes, expiration_days, linked_person_info and so forth. Could you go through them and clean them up? I feel that we can significantly slim down this file.